pkpdapp/pkpdapp/views/dataset.py

```diff
@@ -0,0 +1,200 @@
+#
+# This file is part of PKPDApp (https://github.com/pkpdapp-team/pkpdapp) which
+# is released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from django.shortcuts import render
+from django.urls import reverse_lazy
+from django.core.paginator import Paginator
+from pkpdapp.models import Dataset, Biomarker, BiomarkerType
+from ..forms import CreateNewDataset, CreateNewBiomarkerUnit
+import pandas as pd
+from django.contrib import messages
+from django.apps import apps
+from django.forms import formset_factory
+from django.shortcuts import redirect
+from django.views.generic import (
+    DetailView, CreateView,
+    UpdateView, DeleteView,
+    ListView
+)
+
+
+BASE_FILE_UPLOAD_ERROR = 'FILE UPLOAD FAILED: '
+
+
+class DatasetDetailView(DetailView):
+    model = Dataset
+    paginate_by = 100
+    template_name = 'dataset_detail.html'
+
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get a context
+        context = super().get_context_data(**kwargs)
+        # Add in a QuerySet of all the books
+
+        context['biomarker_types'] = BiomarkerType.objects.filter(
+            dataset=context['dataset']
+        )
+        biomarker_dataset = self.get_paginated_biomarker_dataset(context)
+        context['biomarker_dataset'] = biomarker_dataset
+        context['page_obj'] = biomarker_dataset
+        return context
+
+    def get_paginated_biomarker_dataset(self, context):
+        queryset = Biomarker.objects.filter(
+            biomarker_type__dataset=context['dataset']
+        ).order_by('id')
+        paginator = Paginator(queryset, 20)  # paginate_by
+        page = self.request.GET.get('page')
+        activities = paginator.get_page(page)
+        return activities
+
+
+class DatasetListView(ListView):
+    model = Dataset
+    template_name = 'dataset_list.html'
+
+
+class DatasetCreate(CreateView):
+    model = Dataset
+    fields = ['name', 'description']
+    template_name = 'dataset_form.html'
+
+
+class DatasetUpdate(UpdateView):
+    model = Dataset
+    fields = ['name', 'description']
+    template_name = 'dataset_form.html'
+
+
+def create(request):
```

Comment:
I've been naming the vews as "<ModelName><View>", ie DatasetCreate, and importing them into the views module, so you refer to them as pkpd.views.DatasetCreate. I think its better to make sure all views stick to the same pattern, whatever that is. We probably should stick to either function-based or class-based views as well